From 4ff09c8f6e2fddd7dbbe6e5adb19800db1a5af29 Mon Sep 17 00:00:00 2001
From: Russ Weight <russell.h.weight@intel.com>
Date: Tue, 7 Jun 2016 11:56:08 -0700
Subject: [PATCH 1/1] Revert "tracing: Fix trace_printk() to print when not
 using bprintk()"

This reverts commit 29d5056343389ea5d75677576f30db211001fb58.

This change reverts a patch from the stable kernel that causes
a null-pointer panic.

Change-Id: I78bf89cc50947c6bd5cab9f3a198ab24894222a4
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-33194
Signed-off-by: Russ Weight <russell.h.weight@intel.com>
Reviewed-on: https://android.intel.com:443/513462
---
 include/linux/kernel.h      | 6 +++---
 kernel/trace/trace_printk.c | 3 ---
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/include/linux/kernel.h b/include/linux/kernel.h
index 5eb56c4..0e21563 100644
--- a/include/linux/kernel.h
+++ b/include/linux/kernel.h
@@ -599,7 +599,7 @@ do {							\
 
 #define do_trace_printk(fmt, args...)					\
 do {									\
-	static const char *trace_printk_fmt __used			\
+	static const char *trace_printk_fmt				\
 		__attribute__((section("__trace_printk_fmt"))) =	\
 		__builtin_constant_p(fmt) ? fmt : NULL;			\
 									\
@@ -643,7 +643,7 @@ int __trace_printk(unsigned long ip, const char *fmt, ...);
  */
 
 #define trace_puts(str) ({						\
-	static const char *trace_printk_fmt __used			\
+	static const char *trace_printk_fmt				\
 		__attribute__((section("__trace_printk_fmt"))) =	\
 		__builtin_constant_p(str) ? str : NULL;			\
 									\
@@ -665,7 +665,7 @@ extern void trace_dump_stack(int skip);
 #define ftrace_vprintk(fmt, vargs)					\
 do {									\
 	if (__builtin_constant_p(fmt)) {				\
-		static const char *trace_printk_fmt __used		\
+		static const char *trace_printk_fmt			\
 		  __attribute__((section("__trace_printk_fmt"))) =	\
 			__builtin_constant_p(fmt) ? fmt : NULL;		\
 									\
diff --git a/kernel/trace/trace_printk.c b/kernel/trace/trace_printk.c
index 7b90047..7c8cef6 100644
--- a/kernel/trace/trace_printk.c
+++ b/kernel/trace/trace_printk.c
@@ -292,9 +292,6 @@ static int t_show(struct seq_file *m, void *v)
 	const char *str = *fmt;
 	int i;
 
-	if (!*fmt)
-		return 0;
-
 	seq_printf(m, "0x%lx : \"", *(unsigned long *)fmt);
 
 	/*
-- 
1.9.1

